### PR TITLE
Python packaging workflow

### PR DESCRIPTION
**Pull request by thomasmfish**

### CODE REVIEW

**Review by PeterC-DLS — COMMENTED** [review]
Need to address the linter errors. I have configured the PyPI to use this new workflow.

**Comment by thomasmfish** [comment]
> Need to address the linter errors. I have configured the PyPI to use this new workflow.

Thanks, I have done that now and it appears to be working well. I did try using the python-version matrix but that hits an issue with clashing artifacts per workflow run, so I've gone back to just 3.10. I thought it might be useful to check that it builds across all versions but since only one would be uploaded, so I guess it's not important.

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      contents: write
+
+    steps:
+      # download sdist and wheel from dist job
+      - uses: actions/download-artifact@v4
+      # publish to PyPI using trusted publishing
+      - name: Publish to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          attestations: false
```
Comment (PeterC-DLS) [inline .github/workflows/python-package.yml]: I don't think this is necessary as you don't use relative workflows

Comment (thomasmfish) [reply .github/workflows/python-package.yml]: What do you mean by relative? As in other ones that run as well?

**Comment by PeterC-DLS** [comment]
You could upload only if the version is 3.10 or allow overwrites in the upload action

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (PeterC-DLS) [reply .github/workflows/python-package.yml]: I.e. it uses nested composite actions but the bug may have been fixed with v1.12.1

**Comment by thomasmfish** [comment]
> You could upload only if the version is 3.10 or allow overwrites in the upload action

Okay, I've set it to overwrite, which is passing now. I've also rebased on #4 to avoid potential issues building other versions.

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (thomasmfish) [reply .github/workflows/python-package.yml]: Okay, I've removed that for now and will add it back if the error turns up.

**Comment by thomasmfish** [comment]
Not sure why it was failing for 3.11 and 3.13 before I pushed the last changes ([workflow run](https://github.com/DiamondLightSource/txrm2tiff/actions/runs/11862432713)). It is working now but I didn't do anything that should have affected whether or not it builds...

**Review by PeterC-DLS — APPROVED** [review]
LGTM

**Pull request merged**